src/golang/cmd/server/handler/get_image_url.go

```diff
@@ -99,17 +100,73 @@ func (h *GetImageURLHandler) Perform(ctx context.Context, interfaceArgs interfac
+		conf, err := lib_utils.ParseECRConfig(authConf)
+		if err != nil {
+			return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to parse configuration.")
+		}
+
+		err = container_registry.ValidateECRImage(conf, args.imageName)
+		if err != nil {
+			return emptyResponse, http.StatusUnauthorized, err
+		}
+
+		return getImageURLResponse{
+			Url: fmt.Sprintf("%s/%s", strings.TrimPrefix(conf.ProxyEndpoint, "https://"), args.imageName),
+		}, http.StatusOK, nil
+	} else if args.service == shared.GAR {
+		conf, err := lib_utils.ParseGARConfig(authConf)
+		if err != nil {
+			return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to parse configuration.")
+		}
+
+		// Obtain an OAuth2 token
+		creds, err := google.CredentialsFromJSON(ctx, []byte(conf.ServiceAccountKey), "https://www.googleapis.com/auth/cloud-platform")
+		if err != nil {
+			return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to get credential from service account key.")
+		}
+		token, err := creds.TokenSource.Token()
+		if err != nil {
+			return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to get oauth token.")
+		}
+
+		// Create a new HTTP client
+		client := &http.Client{}
+
+		fullImageUrl := strings.Split(args.imageName, ":")[0]
+		tag := strings.Split(args.imageName, ":")[1]
+
+		host := strings.Split(fullImageUrl, "/")[0]
+		projectID := strings.Split(fullImageUrl, "/")[1]
+		repo := strings.Split(fullImageUrl, "/")[2]
+		image := strings.Split(fullImageUrl, "/")[3]
+
+		// Create a new HTTP request against the Google Artifact Registry API
+		req, err := http.NewRequest("GET", fmt.Sprintf("https://%s/v2/%s/%s/%s/manifests/%s", host, projectID, repo, image, tag), nil)
+		if err != nil {
+			return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to create HTTP request.")
+		}
+
+		// Add the Authorization header to the request
+		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token.AccessToken))
+
+		resp, err := client.Do(req)
+		if err != nil {
+			return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to send request.")
+		}
+
+		if resp.StatusCode == http.StatusOK {
+			return getImageURLResponse{
+				Url: args.imageName,
+			}, http.StatusOK, nil
+		} else if resp.StatusCode == http.StatusUnauthorized || resp.StatusCode == http.StatusForbidden {
+			return emptyResponse, resp.StatusCode, errors.New("Unable to access the requested image. Please double check you have the correct permissions.")
+		} else if resp.StatusCode == http.StatusNotFound {
+			return emptyResponse, resp.StatusCode, errors.New("The requested image was not found.")
+		} else {
+			return emptyResponse, resp.StatusCode, errors.Newf("Received unexpected status: %d", resp.StatusCode)
+		}
+	} else {
+		return emptyResponse, http.StatusBadRequest, errors.Newf("Container registry service %s is not supported.", args.service)
```

Comment:
I may misread, but it seems we will fall at this line if service is GAR?

Reply:
To make it clearer, I removed the `else` statements since we're returning in each case.